attributes/network/cddl/2_ledger.cddl

```diff
@@ -10,49 +10,119 @@ ledger-symbol = address
-    ; specific ledger.
-    4 => {
-        ; TODO: this.
-        ledger-symbol => {
-            0 => tstr,
-            1 => tstr,  ; Description
-            2 => uint, ; Number of precision digits
-            tstr => tstr ; metadata
-        }
-    }
+    ; Binary content of the image.
+    2 => bstr,
+
+    ; Optional height and width, in pixels.
+    ? 3 => [ uint, uint ],
+
+    ; Optional DPI.
+    ? 4 => uint,
+}
+
+; A logo that can be shown visually. The server might add limits to how big
+; a logo can be (both in height x width, and on byte size).
+; A frontend should pick the first one that can be represented in its
+; interface, so this should be an ordered set in decrementing order of
+; importance. For example, a terminal would look for the first unicode character
+; in the set, while a UI on a phone might want a small assets for lists and a
+; big asset for detailed views.
+visual-token-logo = [ * single-visual-token-logo ]
+
+; A summary information for a token. This does not contain extended information
+; which isn't needed in most cases.
+token-info-summary = {
+    ; A human readable name for the token. This is the full name of the token.
+    ; e.g. Bitcoin would use "Bitcoin" here.
+    0 => tstr,
+
+    ; A human readable symbol for the token. This should be _short_ and can
+    ; be limited by the server at creation. This should be unique to the
+    ; ledger.
+    ; e.g. Bitcoin would use "BTC" here.
+    1 => tstr,
+
+    ; Precision decimals. This is used to show how many tokens can be divided.
+    ; It is recommended for the server to limit this number.
+    ; e.g. Bitcoin would use 8.
+    2 => uint,
+
+    ; A transaction fee. A server can deny having no or too low of a transaction
+    ; fee on new tokens or updates. The fee is in the token currency itself.
+    ; Server can still deny transactions based on other criteria (such as
+    ; imposing a ledger's native token as additional transaction fee).
+    ; Servers can also clamp this value to assure a minimum transaction fee.
+    3 => transaction-fee,
+
+    ; A memo, description of the token, information about it that can be
+    ; human readable.
+    ? 4 => memo,
+}
+
+; The supply of a token on a given ledger.
+token-info-supply = {
+    ; The total supply for the token on this particular ledger.
+    ; For Bitcoin, that would be ~19 million at the time of this writing.
+    3 => uint,
+
+    ; The circulating supply for the token on this particular ledger.
+    ; For Bitcoin, that would also be ~19 million, but if there was a staking
+    ; mechanism, that would be less, as those tokens are not burnt but
+    ; frozen.
+    4 => uint,
+
+    ; The maximum supply for the token on this particular ledger.
+    ; This is the maximum amount of tokens at any point in time, including
+    ; in the far future, and should never change.
+    ; If this is null, then the token has an infinite maximum supply.
+    5 => uint / null,
+}
```

Comment:
Why are the indices starting at 3?

Reply:
At first it was merged with summary, but not anymore so removed.